pycon/finaid/views.py

```diff
@@ -425,3 +425,27 @@ def get_value(name, object):
         writer.writerow(data)
 
     return response
+
+
+@login_required
+def receipt_upload(request):
+    if request.method == 'POST':
+        form = ReceiptForm(request.POST, request.FILES)
+
+        if form.is_valid():
+            form.instance.user = request.user
+            form.instance.application = request.user.financial_aid
+            form.save()
+
+            # Display a message to user
+            messages.add_message(request, messages.INFO,
+                                _(u"Receipt submitted"))
+            return redirect("receipt_upload")
+
+    else:
+        form = ReceiptForm()
+        receipts = request.user.financial_aid.receipts.all()
+    return render(request, "finaid/receipt_upload.html", {
+        'form': form,
+        'receipts': receipts
```

Comment:
If the method is POST but the form is not valid, `receipts` never gets set.  I think you can just outdent it once so it runs before the render regardless of the request method.